src/app/reporting/Engine.h

```diff
@@ -102,6 +104,17 @@ class Engine
      */
     static void Run(System::Layer * aSystemLayer, void * apAppState);
 
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+    class ScopedLock
```

Comment:
Is it possible to create a seperate PR?

We can try:

class ScopedLock {
public:
ScopedLock(Mutex & mutex) : mMutex(mutex) { mutex.Lock(); }
~ScopedLock(Mutex & mutex)  { mutex.Unlock(); }
private:
Mutex & mMutex;
}

Reply:
i can create the seperate one

Reply:
removed,